File: app/src/main/java/ImageAdapter.java
import android.content.Context;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.GridView;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.example.speakeasy.R;

import java.util.ArrayList;
import java.util.List;

public class ImageAdapter extends BaseAdapter {
    private Context context;
    private ImageDatabaseHelper imageDatabaseHelper;//bd
    private List<ImageModel> images;//bd

    private List<Uri> imageUris = new ArrayList<>();
    private final int[] imageIds = {
            R.drawable.ardor,
            R.drawable.dolor,
            R.drawable.fastidio,
            R.drawable.fiebre,
            R.drawable.mareo,
            R.drawable.picazon,
            R.drawable.temblor,

            // Agrega más imágenes aquí
    };
    private final String[] imageTitles = {
            "Ardor",
            "Dolor",
            "Fastidio",
            "Fiebre",
            "Mareo",
            "Picazon",
            "Temblor",

            // Agrega más títulos aquí
    };

    public ImageAdapter(Context context) {
        this.context = context;
        this.imageDatabaseHelper = new ImageDatabaseHelper(context);//bd
        this.images = imageDatabaseHelper.getAllImages();//bd
    }

    @Override
    public int getCount() {
        return imageIds.length;
    }

    @Override
    public Object getItem(int position) {
        return null;
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ImageView imageView;
        TextView textView;
        View gridItemView = convertView;
        if (convertView == null) {
            /*imageView = new ImageView(context);
            imageView.setLayoutParams(new GridView.LayoutParams(200, 200));
            imageView.setScaleType(ImageView.ScaleType.CENTER_CROP);
            imageView.setPadding(16, 16, 16, 16);*/
            LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            gridItemView = inflater.inflate(R.layout.grid_item, parent,false);
        } else {
            gridItemView = convertView;
        }

        //imageView.setImageResource(imageIds[position]);
        imageView = gridItemView.findViewById(R.id.imageView);
        textView = gridItemView.findViewById(R.id.textView1);

        if (position < imageIds.length) {//Se agrega para agregar imagenes manualmente
        imageView.setImageResource(imageIds[position]);
        textView.setText(imageTitles[position]);  // Aquí estableces el título
        } else {//Se agrega para agregar imagenes manualmente
            Uri userImageUri = imageUris.get(position - imageIds.length);
            // Utiliza Glide o Picasso para cargar la imagen desde la URI.
            Glide.with(context).load(userImageUri).into(imageView);
            textView.setText("Imagen de usuario");  // Puedes personalizar el título aquí.
        }
        return gridItemView;
        //return imageView;
    }
    public void agregarImagen(Uri imageUri) {
        imageUris.add(imageUri);
        notifyDataSetChanged(); // Asegura que el adaptador se actualice.
    }

}

File: app/src/main/java/ImageAdapter.kt
import android.content.Context
import android.graphics.BitmapFactory
import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import com.example.speakeasy.R
import java.io.ByteArrayOutputStream
import java.io.IOException
import java.io.InputStream
import android.content.Intent
import com.example.speakeasy.ImageDetailActivity

class ImageAdapter(private val context: Context) : BaseAdapter() {
    private val imageDatabaseHelper: ImageDatabaseHelper // bd
    private var images: List<ImageModel> // bd

    private val imageUris = ArrayList<Uri>()
    private val imageIds = intArrayOf(
        R.drawable.ardor,
        R.drawable.dolor,
        R.drawable.fastidio,
        R.drawable.fiebre,
        R.drawable.mareo,
        R.drawable.picazon,
        R.drawable.temblor
        // Agrega más imágenes aquí
    )
    private val imageTitles = arrayOf(
        "Ardor",
        "Dolor",
        "Fastidio",
        "Fiebre",
        "Mareo",
        "Picazon",
        "Temblor"
        // Agrega más títulos aquí
    )

    init {
        imageDatabaseHelper = ImageDatabaseHelper(context) // bd local
        images = imageDatabaseHelper.getAllImages() // bd local
    }

    override fun getCount(): Int {
        return images.size
    }

    override fun getItem(position: Int): Any {
        return images[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {

        var gridItemView = convertView
        if (convertView == null) {
            val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            gridItemView = inflater.inflate(R.layout.grid_item, parent, false)
        }

        val imageView = gridItemView?.findViewById(R.id.imageView) as ImageView
        val textView = gridItemView.findViewById(R.id.textView1) as TextView
        val image = images[position]

        /*imageView.setImageResource(image.id) // Cambia esto si las imágenes se almacenan de manera diferente
        textView.text = image.description;*/
        val imageBitmap = BitmapFactory.decodeByteArray(image.imageBlob, 0, image.imageBlob.size)
        imageView.setImageBitmap(imageBitmap)
        textView.text = image.description

        // Añadimos el OnClickListener aquí al darle click al primer menu va a esta intent
        gridItemView.setOnClickListener {
            val intent = Intent(context, ImageDetailActivity::class.java).apply {
                putExtra("imageBytes", image.imageBlob)
                putExtra("description", image.description)
            }
            context.startActivity(intent)
        }

        return gridItemView
    }

    fun agregarImagen(imageBytes: ByteArray, description: String) {
        try {
            imageDatabaseHelper.insertImage(imageBytes, description)
            images = imageDatabaseHelper.getAllImages()
            notifyDataSetChanged()

        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    private fun getBytesFromInputStream(inputStream: InputStream): ByteArray {
        val byteBuffer = ByteArrayOutputStream()
        val bufferSize = 1024
        val buffer = ByteArray(bufferSize)
        var len: Int
        while (inputStream.read(buffer).also { len = it } != -1) {
            byteBuffer.write(buffer, 0, len)
        }
        return byteBuffer.toByteArray()
    }

    // Agrega tus funciones adicionales según sea necesario
}
